### PR TITLE
context: reimplement Context with pure rust

### LINKED ISSUE #4: Merge SCon into Context
Currently, SCon is just a new type wrapped a plain C string without public accessibility.  The only meaningful thing to do with SCon is convert it to a Context or plain String. We can merge SCon into Context and let the user get the helpers to extract parts of the security context.

### PR DESCRIPTION
**Pull request by crab2313**
Fixes: #4

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -1,167 +1,97 @@
...
-            }
+impl Context {
+    pub fn new(context: &str) -> Option<Self> {
+        let mut iter = context.split(":");
+        let user = iter.next()?.to_owned();
+        let role = iter.next()?.to_owned();
+        let _type = iter.next()?.to_owned();
+        let range = iter.collect::<Vec<&str>>();
+        if range.len() == 0 {
+            return None;
```
Comment (nibon7) [inline src/context.rs]: this may be a bit incompatible with context_new

Comment (nibon7) [reply src/context.rs]: context_new("system_u:object_r:bin_t") will return a valid context_t and you can use context_range_set to set range later.

Comment (nibon7) [reply src/context.rs]: can test with:
```c
#include <selinux/selinux.h>
#include <selinux/context.h>
#include <stdio.h>
#include <assert.h>

int main()
{
        context_t con;
        int ret;
        char *s;

//      con = context_new("system_u:");
        con = context_new("system_u:object_r:bin_t");
        assert(con != NULL);
        //con= context_new("system_u:object_r:bin_t:s0");

        printf("user: %s\n", context_user_get(con));
        printf("role: %s\n", context_role_get(con));
        printf("type: %s\n", context_type_get(con));
        printf("range: %s\n", context_range_get(con));

        context_range_set(con, "s0");
        printf("range: %s\n", context_range_get(con));

        s = context_str(con);

        ret = security_check_context(s);

        printf("ret=%d\n", ret);

        context_free(con);
        return 0;
}
```

**Review by nibon7 — CHANGES_REQUESTED** [review]

File: src/context.rs (hunk 1 above)
Comment (crab2313) [reply src/context.rs]: Is `range` the only field that can be optional?

Comment (crab2313) [reply src/context.rs]: Issue addressed.

**Review by nibon7 — APPROVED** [review]

**Pull request merged**